lang/cs/Org.Apache.REEF.IMRU/OnREEF/MapInputWithControlMessage/MapInputWithControlMessageCodec.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Wake.Remote;
+using Org.Apache.REEF.Wake.StreamingCodec;
+
+namespace Org.Apache.REEF.IMRU.OnREEF.MapInputWithControlMessage
+{
+    /// <summary>
+    /// Streaming codec for MapInputWithControlMessage
+    /// </summary>
+    /// <typeparam name="TMapInput"></typeparam>
+    internal class MapInputWithControlMessageCodec<TMapInput> : IStreamingCodec<MapInputWithControlMessage<TMapInput>>
+    {
+        private readonly IStreamingCodec<TMapInput> _baseCodec;
+
+        [Inject]
+        private MapInputWithControlMessageCodec(IStreamingCodec<TMapInput> baseCodec)
+        {
+            _baseCodec = baseCodec;
+        }
+
+        /// <summary>
+        /// Reads message from reader
+        /// </summary>
+        /// <param name="reader">reader from which to read the message</param>
+        /// <returns>Read message</returns>
+        MapInputWithControlMessage<TMapInput> IStreamingCodec<MapInputWithControlMessage<TMapInput>>.Read(
+            IDataReader reader)
+        {
+            string controlMessageString = reader.ReadString();
```

Comment:
Yes makes sense. Lemme do that.

On Thu, Aug 6, 2015 at 10:10 AM, Markus Weimer <notifications@github.com>
wrote:

> In
> lang/cs/Org.Apache.REEF.IMRU/OnREEF/MapInputWithControlMessage/MapInputWithControlMessageCodec.cs
> <https://github.com/apache/incubator-reef/pull/338#discussion_r36439597>:
>
> > +
> > +        [Inject]
> > +        private MapInputWithControlMessageCodec(IStreamingCodec<TMapInput> baseCodec)
> > +        {
> > +            _baseCodec = baseCodec;
> > +        }
> > +
> > +        /// <summary>
> > +        /// Reads message from reader
> > +        /// </summary>
> > +        /// <param name="reader">reader from which to read the message</param>
> > +        /// <returns>Read message</returns>
> > +        MapInputWithControlMessage<TMapInput> IStreamingCodec<MapInputWithControlMessage<TMapInput>>.Read(
> > +            IDataReader reader)
> > +        {
> > +            string controlMessageString = reader.ReadString();
>
> This is potentially a long string. Can we encode them into int or, better
> yet, byte?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-reef/pull/338/files#r36439597>.
>